sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -217,9 +222,14 @@ private[sql] case class DataSourceScan(
     // TODO: The abstractions between this class and SqlNewHadoopRDD makes it difficult to know
     // here which path to use. Fix this.
 
-    ctx.INPUT_ROW = row
     ctx.currentVars = null
-    val columns1 = exprs.map(_.gen(ctx))
+    val colVars = output.zipWithIndex.map(x => ctx.freshName("col" + x._2))
+    val colDeclarations = colVars.zipWithIndex.map(x => {
+      (columnVectorClz + " " + x._1 + " = " + batch + ".column(" + x._2 + ");\n").trim} )
```

Comment:
@davies , this comment makes sense.
Since the class [```ColumnarBatch```](https://github.com/apache/spark/blob/f96997ba244a14c26e85a2475415a762d0c0d0a8/sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnarBatch.java#L47) is declared as final, I expect a method call ```batch.column()``` is inlined as ```batch.columns[]``` by a Java JIT compiler.
Or, is it better to explicitly access ```batch.columns[]``` by making ```ColumnarBatch.columns[]``` public?